### PR TITLE
Fix(eos_designs): Fix schema validation of dynamic keys

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Fix schema validation of dynamic keys

The schema validation tooling did not have a way of populating the default values of dynamic keys lists like `node_type_keys`, so this was implemented manually before running validation. 

The default values were loaded from `shared_utils` but since no validation had been done, `shared_utils` would fail badly if any input was not as expected.

### CODE REVIEW

**Review by emilarista — APPROVED** [review]
Tested with repo that broke node_type_keys, connected_endpoints_keys and network_services_keys. All faulty variables now result in proper schema validation type errors. LGTM!

File: python-avd/schema_tools/metaschema/meta_schema_model.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ class ConvertType(str, Enum):
     """Default value"""
     valid_values: list[bool] | None = None
     """List of valid values"""
-    dynamic_valid_values: str | None = None
+    dynamic_valid_values: list[str] | None = None
     """
     Path to variable under the parent dictionary containing valid values.
```
Comment (gmuloc) [inline python-avd/schema_tools/metaschema/meta_schema_model.py]: Should the description change? It is now a list of paths?

File: python-avd/pyavd/_schema/utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from collections import ChainMap
+
+from pyavd._utils import get
+
+
+def get_instance_with_defaults(instance: dict, dynamic_key_path: str, schema: dict) -> dict | ChainMap:
```
Comment (gmuloc) [inline python-avd/pyavd/_schema/utils.py:9]: probably worth adding a docstring here

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/connected-endpoints.md [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
     | Variable | Type | Required | Default | Value Restrictions | Description |
     | -------- | ---- | -------- | ------- | ------------------ | ----------- |
-    | [<samp>&lt;connected_endpoints_keys.key&gt;</samp>](## "<connected_endpoints_keys.key>") | List, items: Dictionary |  | See (+) on YAML tab |  | This should be applied to group_vars or host_vars where endpoints are connecting.<br>`connected_endpoints_keys.key` is one of the keys under "connected_endpoints_keys".<br>The default keys are `servers`, `firewalls`, `routers`, `load_balancers`, and `storage_arrays`.<br> |
+    | [<samp>&lt;connected_endpoints_keys.key&gt;</samp>](## "<connected_endpoints_keys.key>") | List, items: Dictionary |  |  |  | This should be applied to group_vars or host_vars where endpoints are connecting.<br>`connected_endpoints_keys.key` is one of the keys under "connected_endpoints_keys".<br>The default keys are `servers`, `firewalls`, `routers`, `load_balancers`, and `storage_arrays`.<br> |
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/tables/connected-endpoints.md]: is the default still correct? (in description) it seems to be misssin some (camera. phones...) maybe best to remove it and make sure we have the default keys somewhere else. Also we have a static list for these in the md... maybe we could read it from the default on the connected_endpoint_keys

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/docs/tables/connected-endpoints.md]: Removed the list

File: python-avd/pyavd/_schema/utils.py (hunk 2 above)
Comment (ClausHolbechArista) [reply python-avd/pyavd/_schema/utils.py:9]: added

File: python-avd/schema_tools/metaschema/meta_schema_model.py (hunk 1 above)
Comment (ClausHolbechArista) [reply python-avd/schema_tools/metaschema/meta_schema_model.py]: fixed

File: python-avd/schema_tools/metaschema/meta_schema_model.py [hunk 4]
```diff
@@ -339,7 +339,7 @@ def __str__(self) -> str:
     """
     valid_values: list[str] | None = None
     """List of valid values"""
-    dynamic_valid_values: str | None = None
+    dynamic_valid_values: list[str] | None = None
     """
     Path to variable under the parent dictionary containing valid values.
```
Comment (gmuloc) [inline python-avd/schema_tools/metaschema/meta_schema_model.py]:
```suggestion
    List of paths to variable under the parent dictionary containing valid values.
```

File: python-avd/schema_tools/metaschema/meta_schema_model.py [hunk 5]
```diff
@@ -242,7 +242,7 @@ class ConvertType(str, Enum):
     """Maximum value"""
     valid_values: list[int] | None = None
     """List of valid values"""
-    dynamic_valid_values: str | None = None
+    dynamic_valid_values: list[str] | None = None
     """
     Path to variable under the parent dictionary containing valid values.
```
Comment (gmuloc) [inline python-avd/schema_tools/metaschema/meta_schema_model.py]:
```suggestion
    List of path to variable under the parent dictionary containing valid values.
```

**Review by gmuloc — APPROVED** [review]

**Pull request merged**